core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForShort.java

```diff
@@ -192,12 +192,24 @@ private void compressDataMyOwnWay(ColumnWithShortIndex[] indexes) {
     }
     map.add(start);
     map.add(counter);
-    this.keyBlock = convertToKeyArray(list);
-    if (indexes.length == keyBlock.length) {
-      dataIndexMap = new short[0];
-    } else {
+    boolean useRle = (list.size() > indexes.length
```

Comment:
I guess you can simply as below.
```
boolean useRle = !((((list.size() + map.size()) * 100) / indexes.length) > 70);
```
I think need of `list.size() > indexes.length` is not required as the percentage calculation can include this as well.